File: protoc-http-go/demo_output/helloworld.go
package helloworld

import (
	"bytes"
	"context"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

// HelloRequest represents the HelloRequest message from the proto definition
type HelloRequest struct {
	Name string `json:"name"`
}

// HelloReply represents the HelloReply message from the proto definition
type HelloReply struct {
	Message string `json:"message"`
}

// GreeterClient is an HTTP client for the Greeter service
type GreeterClient struct {
	BaseURL    string
	HTTPClient *http.Client
}

// NewGreeterClient creates a new GreeterClient with the given base URL
func NewGreeterClient(baseURL string) *GreeterClient {
	return &GreeterClient{
		BaseURL:    baseURL,
		HTTPClient: &http.Client{},
	}
}

// NewGreeterClientWithClient creates a new GreeterClient with a custom HTTP client
func NewGreeterClientWithClient(baseURL string, httpClient *http.Client) *GreeterClient {
	return &GreeterClient{
		BaseURL:    baseURL,
		HTTPClient: httpClient,
	}
}

// SayHello calls the SayHello RPC method
func (c *GreeterClient) SayHello(ctx context.Context, req *HelloRequest) (*HelloReply, error) {
	// Serialize request to JSON
	reqJSON, err := json.Marshal(req)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	// Create HTTP request
	url := c.BaseURL + "/helloworld/say-hello"
	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(reqJSON))
	if err != nil {
		return nil, fmt.Errorf("failed to create HTTP request: %w", err)
	}

	// Set headers
	httpReq.Header.Set("Content-Type", "application/json")
	httpReq.Header.Set("Accept", "application/json")

	// Make HTTP request
	resp, err := c.HTTPClient.Do(httpReq)
	if err != nil {
		return nil, fmt.Errorf("HTTP request failed: %w", err)
	}
	defer resp.Body.Close()

	// Read response body
	respBody, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, fmt.Errorf("failed to read response body: %w", err)
	}

	// Check status code
	if resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("HTTP request failed with status %d: %s", resp.StatusCode, string(respBody))
	}

	// Deserialize response
	var response HelloReply
	if err := json.Unmarshal(respBody, &response); err != nil {
		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
	}

	return &response, nil
}

File: protoc-http-go/demo_output/helloworld.vb
Option Strict On
Option Explicit On
Option Infer On

Imports System
Imports System.Net.Http
Imports System.Net.Http.Headers
Imports System.Threading
Imports System.Threading.Tasks
Imports System.Text
Imports System.Collections.Generic
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Serialization

Namespace Helloworld

' HelloReply represents the HelloReply message from the proto definition
Public Class HelloReply
    <JsonProperty("message")>
    Public Property Message As String
End Class

' HelloRequest represents the HelloRequest message from the proto definition
Public Class HelloRequest
    <JsonProperty("name")>
    Public Property Name As String
End Class

' GreeterClient is an HTTP client for the Greeter service
Public Class GreeterClient
    Public Property BaseUrl As String
    Private ReadOnly _httpClient As HttpClient

    Public Sub New(baseUrl As String)
        Me.BaseUrl = baseUrl
        Me._httpClient = New HttpClient()
    End Sub

    Public Sub New(baseUrl As String, httpClient As HttpClient)
        Me.BaseUrl = baseUrl
        Me._httpClient = httpClient
    End Sub

    Private Async Function PostJsonAsync(Of TResponse)(url As String, requestBody As Object, cancellationToken As CancellationToken) As Task(Of TResponse)
        Dim settings As New JsonSerializerSettings() With { .ContractResolver = New CamelCasePropertyNamesContractResolver() }
        Dim reqJson As String = JsonConvert.SerializeObject(requestBody, settings)
        Using httpRequest As New HttpRequestMessage(HttpMethod.Post, url)
            httpRequest.Content = New StringContent(reqJson, Encoding.UTF8, "application/json")
            httpRequest.Headers.Accept.Clear()
            httpRequest.Headers.Accept.Add(New MediaTypeWithQualityHeaderValue("application/json"))
            Dim response As HttpResponseMessage = Await _httpClient.SendAsync(httpRequest, cancellationToken)
            Dim respBody As String = Await response.Content.ReadAsStringAsync()
            If Not response.IsSuccessStatusCode Then
                Throw New HttpRequestException(String.Format("HTTP request failed with status {0}: {1}", CInt(response.StatusCode), respBody))
            End If
            Dim result As TResponse = JsonConvert.DeserializeObject(Of TResponse)(respBody, settings)
            Return result
        End Using
    End Function

    ' SayHelloAsync calls the SayHello RPC method
    Public Async Function SayHelloAsync(request As HelloRequest, Optional cancellationToken As CancellationToken = Nothing) As Task(Of HelloReply)
        Dim url As String = Me.BaseUrl & "/helloworld/say-hello/" & "v1"
        Return Await PostJsonAsync(Of HelloReply)(url, request, cancellationToken)
    End Function

End Class

End Namespace
